ns/adapters/outbound/smtp_client.py

```diff
@@ -31,23 +31,25 @@ class SmtpClientConfig(BaseSettings):
     smtp_port: int = Field(..., description="The port for the mail server connection")
     login_user: str = Field(..., description="The login username or email")
     login_password: str = Field(..., description="The login password")
+    use_starttls: bool = Field(
+        default=True, description="Boolean flag indicating the use of STARTTLS"
+    )
 
 
 class SmtpClient(SmtpClientPort):
     """Concrete implementation of an SmtpClientPort"""
 
-    def __init__(self, config: SmtpClientConfig, debugging: bool = False):
+    def __init__(self, config: SmtpClientConfig):
         """Assign config, which should contain all needed info"""
         self._config = config
-        self._debugging = debugging
 
     def send_email_message(self, message: EmailMessage):
         # create ssl security context per Python's Security considerations
         context = ssl.create_default_context()
```

Comment:
I think this should be moved into the "if" block below.

Reply:
I don't get what you mean, instead of 'debugging' we are now using 'use_starttls' from config, so I believe we don't need 'debugging' anymore.

Reply:
Sorry, looks like I tagged the wrong line. I meant the line `context = ssl.create_default_context()`.

Reply:
You are right the context is only used when using starttls. I moved it into the if block.